editor/gui/scene_tree_editor.cpp

```diff
@@ -1684,13 +1684,19 @@ void SceneTreeDialog::_show_all_nodes_changed(bool p_button_pressed) {
 }
 
 void SceneTreeDialog::set_valid_types(const Vector<StringName> &p_valid) {
-	if (p_valid.is_empty()) {
-		return;
+	if (allowed_types_hbox) {
+		allowed_types_hbox->queue_free();
+		valid_type_icons.clear();
 	}
 
 	tree->set_valid_types(p_valid);
 
+	if (p_valid.is_empty()) {
+		return;
+	}
+
 	HBoxContainer *hbox = memnew(HBoxContainer);
+	allowed_types_hbox = hbox;
```

Comment:
```suggestion
	allowed_types_hbox = memnew(HBoxContainer);
```
You could just do that, although it requires more changes.

Reply:
I've swapped hbox for allowed_types_hbox.